coveralls/api.py

```diff
@@ -76,67 +72,78 @@ def ensure_token(self):
     @staticmethod
     def load_config_from_appveyor():
         pr = os.environ.get('APPVEYOR_PULL_REQUEST_NUMBER')
-        return 'appveyor', os.environ.get('APPVEYOR_BUILD_ID'), pr
+        return 'appveyor', os.environ.get('APPVEYOR_BUILD_ID'), None, pr
 
     @staticmethod
     def load_config_from_buildkite():
         pr = os.environ.get('BUILDKITE_PULL_REQUEST')
         if pr == 'false':
             pr = None
-        return 'buildkite', os.environ.get('BUILDKITE_JOB_ID'), pr
+        return 'buildkite', os.environ.get('BUILDKITE_JOB_ID'), None, pr
 
     @staticmethod
     def load_config_from_circle():
         pr = os.environ.get('CI_PULL_REQUEST', '').split('/')[-1] or None
-        return 'circle-ci', os.environ.get('CIRCLE_BUILD_NUM'), pr
+        return 'circle-ci', os.environ.get('CIRCLE_BUILD_NUM'), None, pr
 
-    @staticmethod
-    def load_config_from_github():
-        service_number = os.environ.get('GITHUB_SHA')
+    def load_config_from_github(self):
+        service = 'github'
+        if self.config.get('repo_token'):
+            service = 'github-actions'
+        else:
+            gh_token = os.environ.get('GITHUB_TOKEN')
+            if not gh_token:
+                raise CoverallsException(
+                    'Running on Github Actions but GITHUB_TOKEN is not set. '
+                    'Add "env: GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}" to '
+                    'your step config.')
+            self.config['repo_token'] = gh_token
+
+        number = os.environ.get('GITHUB_RUN_NUMBER')
```

Comment:
This should really be `GITHUB_RUN_ID` in case anyone wants multiple different github actions workflows reporting to coveralls. Also, using `GITHUB_RUN_ID` makes it possible to use https://github.com/marketplace/actions/coveralls-github-action to finalize the parallel build.

Basically, neither of the two is unique enough to be used as `job`, but `GITHUB_RUN_ID` is better to identify the build (workflow run), as it's used in github actions URLs, etc.